lexer/lexer.go

```diff
@@ -193,48 +212,68 @@ func (l *Lexer) eatWhitespace() {
 	}
 }
 
-func (l *Lexer) eatUntil(untilChar byte) {
+// TODO: This should panic if it doesn't find *untilChar* and reaches EOF
+func (l *Lexer) eatUntil(untilChar rune) string {
+	var eatenStr string
+
 	for l.CurrentChar() != untilChar {
-		l.NextChar()
+		char := l.NextChar()
+		// fmt.Printf("  (eat:%d) %q\n", l.index, char)
+
+		eatenStr = eatenStr + string(char)
 	}
 
-	// Point to the byte after our until
+	return eatenStr
+}
+
+func (l *Lexer) eatUntilAfter(untilChar rune) string {
+	eatenStr := l.eatUntil(untilChar)
+
+	// Point to the byte after our untilChar
 	l.NextChar()
+
+	return eatenStr
 }
 
 func (l *Lexer) eatCurrentLine() {
-	l.eatUntil('\n')
+	l.eatUntilAfter('\n')
 }
 
+// TODO...
 func (l *Lexer) eatMultiLineComment() {
 }
 
 func (l *Lexer) lexIdentifier() (token.Token, error) {
+	var char rune
 	var endIndex int
 
-	reachedEnd := false
 	startIndex := l.index
 
-	for isIdentifier(rune(l.CurrentChar())) {
-		_, err := l.NextChar()
+	for isIdentifier(l.CurrentChar()) {
+		char = l.NextChar()
+
+		// fmt.Printf("  (ident:%d) %q\n", l.index-1, char)
 
-		// Probably because we have reached the end of the source string
-		// TODO: Check for special EOF to make sure this is the case
-		if err != nil {
-			reachedEnd = true
+		if char == EOF {
 			break
 		}
 	}
 
-	// For half-open intervals
-	if reachedEnd {
+	// Because go has half-open ranges we need to go one further than the end index to get the full
+	// identifier
+	if l.reachedEnd {
```

Comment:
wondering if Peek or NextChar should also return a bool indicating if the end has reached, similar to when reading from go channels. Then this could become:

```
if _, ok := l.Peek {
}
```

or we could save the value from the NextChar call above also